src/lib/runtime/Web3FunctionRunner.ts

```diff
@@ -322,7 +350,13 @@ export class Web3FunctionRunner {
     context: Web3FunctionContextData,
     options: Web3FunctionRunnerOptions,
     multiChainProviderConfig: MultiChainProviderConfig
-  ): Promise<{ result: Web3FunctionResult; storage: Web3FunctionStorage }> {
+  ): Promise<{
+    data: {
+      result: Web3FunctionResult | undefined;
+      storage: Web3FunctionStorage;
+    };
+    callbacks: Web3FunctionCallbackStatus;
```

Comment:
I'd rather avoid having return data at different nested levels, seem cleaner to have:
```suggestion
  ): Promise<{
    result: Web3FunctionResult | undefined;
    storage: Web3FunctionStorage;
    callbacks: Web3FunctionCallbackStatus;
```